federation/ldap/src/main/java/org/keycloak/storage/ldap/LDAPStorageProvider.java

```diff
@@ -450,12 +523,20 @@ protected List<LDAPObject> searchLDAP(RealmModel realm, Map<String, String> attr
                 Condition usernameCondition = conditionsBuilder.equal(UserModel.USERNAME, attributes.get(UserModel.USERNAME), EscapeStrategy.NON_ASCII_CHARS_ONLY);
                 ldapQuery.addWhereCondition(usernameCondition);
 
-                List<LDAPObject> ldapObjects = ldapQuery.getResultList();
+                //need to handle offset manually as there has to be results collected first due to possible duplicate resutls from different fields
+                List<LDAPObject> ldapObjects = paginatedSearchLDAP(ldapQuery, null, ldapMaxResult);
+                
                 results.addAll(ldapObjects);
                 unique.addAll(ldapObjects);
             }
         }
 
+        // if ldapMaxResult == null -> no pagination
+        // if result has already enough values return
+        if (ldapMaxResult != null && ldapMaxResult.compareTo(results.size()) <= 0) {
+            return paginateResults(results, firstResult, maxResults);
+        }
```

Comment:
Should we decrease `ldapMaxResults` here?

Reply:
Theoretically it might be decreased. I'd say not exactly `ldapMaxResults` because it determines that there is collected enough objects to cover also for offset, but for this call https://github.com/keycloak/keycloak/pull/19050/files#diff-bb4ffa08fae23dc22710d174b81fadf5028b58a8742d9fba4035a2c916e92982R549 it might. But I decided not to due to a fact that some results could be duplicated in the results and it may lead to a situation when it ends up with less objects than required.

Reply:
This is now outdated.